pkg/version/version.go

```diff
@@ -199,3 +216,106 @@ func GetForwardedPortsFromAppSpec(appID string, sequence int64) ([]types.Forward
 
 	return ports, nil
 }
+
+func getDownstreamStatus(appID string, sequence int64) (storetypes.DownstreamVersionStatus, error) {
+	db := persistence.MustGetDBSession()
+	var status sql.NullString
+	query := `
+SELECT
+	adv.status
+FROM
+	app_downstream_version AS adv
+WHERE
+	adv.app_id = $1 AND
+	adv.parent_sequence = $2`
+	row := db.QueryRow(query, appID, sequence)
+	if err := row.Scan(
+		&status,
+	); err != nil {
+		return "", errors.Wrap(err, "failed to scan")
+	}
+	return storetypes.DownstreamVersionStatus(status.String), nil
+}
+
+func isBlockedDueToStricPreFlights(appID string, sequence int64) (bool, error) {
```

Comment:
`isBlockedDueToStrictPreFlights` is doing more than fetching stored data
if status is `pending_preflight`, it will poll for status every 2 seconds till it is changed to `pending`
and analyze the preflight result and validate if preflight has strict and preflight result are not failing for strict.

....
and after writing the above, i realize that I could reuse `func (s *KOTSStore) GetCurrentVersion` and do the same 🤦🏽

Reply:
I could have re-used existing store funcs, changing the code to re-use store funcs